### DISCUSSION TITLE
How to conditonally select columns in queryRaw

### DISCUSSION THREAD

**Discussion by amirhhashemi** [prisma/prisma#9006] [Q&A]
I want to conditionally select "password" and "tokenVersion" based on "withPasswordAndTokenVersion" variable.
this is my query:
```ts
    const user = await this.prisma.$queryRaw`
        SELECT
            id,
            name,
            username,
            avatar,
            bio,
            verified,
            "isAdmin",
            ${withPasswordAndTokenVersion && "password"},
            ${withPasswordAndTokenVersion && "tokenVersion"}
        FROM "User"
        `
```
and this is the result. 
```js
[ 
  { 
    id: 1, 
    name: 'arman',
    username: 'cck85379', 
    avatar: 'e03fe005-ee60-4e52-a027-cbc64fc49808.png', 
    bio: null, 
    verified: true, 
    isAdmin: false,
    '?column?': 'tokenVersion' // what is this
  }
] 
```

I think Prisma tries to scape those two lines or something similar because they are in a template string but I don't know how to solve it.

**Comment by ryands17** [comment]
@ahhshm 👋 

Have a look at this: https://github.com/blakeembrey/sql-template-tag/issues/17

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
